src/App.tsx

```diff
@@ -1,61 +1,148 @@
 import './App.scss';
+import { useState } from 'react';
+import { TodoList } from './components/TodoList';
+import { Todo } from './types/Todo';
+import { User } from './types/User';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+function getUserById(userId: number): User | null {
+  const foundUser = usersFromServer.find(({ id }) => id === userId);
+
+  return foundUser || null;
+}
+
+const todos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUserById(todo.userId),
+}));
+
+const getNewTodoId = (visibleTodos: Todo[]) => {
+  return Math.max(...visibleTodos.map(todo => todo.id)) + 1;
+};
+
+type Element = HTMLInputElement | HTMLSelectElement;
 
 export const App = () => {
+  const [titleOfTodo, setTitle] = useState('');
+  const [selectedUser, setSelectedUser] = useState<User | null>(null);
+  const [preparedTodos, setTodos] = useState(todos);
+  const [isUserError, setIsUserError] = useState(false);
+  const [isTitleError, setIsTitleError] = useState(false);
+
+  const handleChange = (event: React.ChangeEvent<Element>) => {
+    const { id, value } = event.target;
+
+    switch (id) {
+      case 'titleInputForm':
+        setIsTitleError(false);
+        setTitle(value);
+        break;
+
+      case 'userSelectForm':
+        setIsUserError(false);
+        setSelectedUser(getUserById(+(value)));
+        break;
+
+      default:
+        throw new Error('Errrror');
+    }
+  };
+
+  const handleAdd = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (!selectedUser) {
+      setIsUserError(true);
+    }
+
+    if (!titleOfTodo.trim()) {
```

Comment:
move it to the variable
```suggestion
    if (...) {
```